src/lib/redux/siteData.ts

```diff
@@ -19,19 +22,32 @@ const refreshSiteData = async (loadSilently: boolean): Promise<void> => {
 
   const siteData = {
     sites,
-    currentSite,
-  };
+    currentSite
+  }
+
+  const siteIdsToCustomers = extractCustomerDataFromSite(sites);
+
+  store.dispatch(saveCustomerData({ siteIdsToCustomers }));
   store.dispatch(saveSiteData(siteData));
 };
 
+const extractCustomerDataFromSite = (sites: SiteRecord[]): Record<SiteId, CustomerRecord[]> => {
+
+  const siteIdsToCustomers: Record<SiteId, CustomerRecord[]> = {};
+  sites.map((site: SiteRecord) => {
+    siteIdsToCustomers[site.id] = site.customers || [];
+    delete site.customers;
```

Comment:
nit: add a comment here - deleting makes me nervous in general lol, but AFAICT when necessary you need to use `getAllCustomersInSite` instead for the main immutability issue etc. I think it'd be useful to add a note inline